crates/lgn-api-codegen/src/rust/snapshots/lgn_api_codegen__rust__tests__rust_generation.snap

```diff
@@ -2151,6 +2151,62 @@ pub mod components {
         Yellow,
     }
 
+    impl std::str::FromStr for CarColor {
+        type Err = Error;
+
+        fn from_str(s: &str) -> Result<Self, <CarColor as std::str::FromStr>::Err> {
+            match s {
+                "Red" => Ok(Self::Red),
+                "Blue" => Ok(Self::Blue),
+                "Yellow" => Ok(Self::Yellow),
+                _ => Err(Error::Internal(format!("unknown enum value {}", s))),
+            }
+        }
+    }
+
+    impl TryFrom<u32> for CarColor {
+        type Error = Error;
+
+        fn try_from(n: u32) -> Result<Self, <CarColor as TryFrom<u32>>::Error> {
+            match n {
+                0 => Ok(Self::Red),
+                1 => Ok(Self::Blue),
+                2 => Ok(Self::Yellow),
+                _ => Err(Error::Internal(format!(
+                    "index out of bound {}, accepted index between 0 and 2",
+                    n
+                ))),
+            }
+        }
+    }
+
+    impl TryFrom<i32> for CarColor {
+        type Error = Error;
+
+        fn try_from(n: i32) -> Result<Self, <CarColor as TryFrom<i32>>::Error> {
+            match n {
+                0 => Ok(Self::Red),
+                1 => Ok(Self::Blue),
+                2 => Ok(Self::Yellow),
+                _ => Err(Error::Internal(format!(
+                    "index out of bound {}, accepted index between 0 and 2",
+                    n
+                ))),
+            }
+        }
+    }
+
+    impl ToString for CarColor {
+        fn to_string(&self) -> String {
+            match self {
+                Self::Red => "Red",
+                Self::Blue => "Blue",
+                Self::Yellow => "Yellow",
+            }
+            .into()
+        }
+    }
```

Comment:
@ademange-legion  @ereOn I added some impl for our enums (ToString, FromStr, From<u32>, and From<i32>), they can be handy when it comes to converting existing values. Let me know if you want to drop them or add an external crate for that 👍